exonum/src/lib.rs

```diff
@@ -12,52 +12,101 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Exonum blockchain framework.
+//! Core library for the Exonum blockchain framework.
 //!
-//! For more information see the project readme.
+//! Exonum is an extensible open-source framework for
+//! creating blockchain applications. Exonum can be used to create cryptographically
+//! powered distributed ledgers in virtually any problem domain, including finance,
+//! governance, and legal. The Exonum framework is oriented towards creating
+//! permissioned blockchains, that is, blockchains with the known set of blockchain
+//! infrastructure providers.
+//!
+//! For more information about the framework see the [readme] and the [Exonum website].
+//!
+//! [readme]: https://github.com/exonum/exonum#readme
+//! [Exonum website]: https://exonum.com/
+//!
+//! # Crate Overview
+//!
+//! This crate provides the tools necessary to work with the Exonum framework.
```

Comment:
Not sure about the phrasing. This crate not only provides "tools", but it contains the actual logic of the blockchain mechanism (excluding consensus) and guarantees for the smart contracts environments.